sway-core/src/semantic_analysis/ast_node/expression/typed_expression.rs

```diff
@@ -282,6 +297,14 @@ impl DeterministicallyAborts for TypedExpression {
             AbiName(_) => false,
             EnumTag { exp } => exp.deterministically_aborts(),
             UnsafeDowncast { exp, .. } => exp.deterministically_aborts(),
+            Reassignment(_reassignment) => {
+                // FIXME: This seems very wrong. We should check rhs, no?
+                false
+            }
+            StorageReassignment(_storage_reassignment) => {
+                // FIXME: This seems very wrong. We should check rhs, no?
+                false
```

Comment:
This is in `fn deterministically_aborts`. We always return `false` for assignments, but if the rhs of an assignment deterministically aborts then the whole expression does. So we should check rhs here, yeah?

Reply:
yeah, technically the RHS could be a code block with a return in it, so we should check that.